### PR TITLE
Feature/entity dependencies card enhancements

### PR DESCRIPTION
**Pull request by Abhay-soni-developer**
I have extended the EntityHasSubcomponentsCard, EntityDependsOnResourcesCard and EntityDependsOnResourcesCard and EntityTable to supports tableOptions.

i have extended these components in such a way that it does not effect the current implementation just extend it capabilities to support tableOptions

This PR is the implementation for this https://github.com/backstage/backstage/issues/18503

### CODE REVIEW

**Comment by Abhay-soni-developer** [comment]
@Rugvip  @benjaminr  @Edje-C @awanlin

**Comment by NishkarshRaj** [comment]
This is pretty helpful.

As we scale Backstage adoption, we don't want to scroll down a lot. So from a UX perspective - Pagination and Filtering is a bliss.

**Comment by NishkarshRaj** [comment]
> ## Missing Changesets
> The following package(s) are changed by this PR but do not have a changeset:
> 
> * **@backstage/plugin-catalog-react**
> * **@backstage/plugin-catalog**
> 
> See [CONTRIBUTING.md](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets) for more information about how to add changesets.
> 
> ## Changed Packages
> Package Name	Package Path	Changeset Bump	Current Version
> @backstage/plugin-catalog-react	plugins/catalog-react	**none**	`v1.7.1-next.0`
> @backstage/plugin-catalog	plugins/catalog	**none**	`v1.11.3-next.0`

@Abhay-soni-developer ☝️

**Comment by benjdlambert** [comment]
@Abhay-soni-developer this will need some changesets :pray:

File: .changeset/popular-icons-doubt.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog-react': minor
+'@backstage/plugin-catalog': minor
+---
+
+I have extended the EntityHasSubcomponentsCard, EntityDependsOnResourcesCard, EntityDependsOnResourcesCard and EntityTable to support table options.
```
Comment (benjdlambert) [inline .changeset/popular-icons-doubt.md:6]:
```suggestion
Tables which use `EntityTableProps` now have an additional `tableOptions` prop which can be used to provide additional table options to these components.
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]

**Comment by Abhay-soni-developer** [comment]
@benjdlambert  the requested changes  have been done. are we ready to merge this

**Review by benjdlambert — APPROVED** [review]

**Comment by Abhay-soni-developer** [comment]
@benjdlambert  i just merged the feedbacks after that pipeline is failing , can you help me resolve this
as i can not find the reason of failure

**Comment by benjdlambert** [comment]
The e2e's are failing for a seperate reason - i've just mer7ged this.

**Pull request merged**